src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java

```diff
@@ -0,0 +1,51 @@
+package com.epam.reportportal.serializers;
+
+import com.epam.reportportal.exception.InternalReportPortalClientException;
+import com.epam.reportportal.restendpoint.http.exception.SerializerException;
+import com.epam.reportportal.restendpoint.serializer.Serializer;
+import com.google.common.net.MediaType;
+import com.google.common.reflect.TypeToken;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.lang.reflect.Type;
+
+public class NotJsonSerializer implements Serializer {
+
+	@Override
+	public <T> byte[] serialize(T t) throws SerializerException {
+
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public <T> T deserialize(byte[] content, Class<T> clazz) throws SerializerException {
+
+		throw new InternalReportPortalClientException("Report portal is not functioning correctly. Response is not json");
```

Comment:
i guess the better place for this logic is ErrorHandler, not a serializer. Can we try to move it there?

Reply:
I was going to do the same, but the thing is that as I understood(tried) ErrorHandler is used when server responded with error. In this case server responded with 200 status but data was wrong, so we need Serializer to handle it. What do you think ?